syntax = "proto3";

package log_streaming;

service LogStreamingService {
    rpc StreamLogEntries (LogStreamRequest) returns (stream LogStreamResponse);
    rpc StopLogStream (StopStreamRequest) returns (StopStreamResponse);
}

message LogStreamRequest {
    string log_file_path = 1;
    repeated string filter_patterns = 2;  // Only stream entries matching these patterns
    int64 start_from_timestamp = 3;       // Unix timestamp to start streaming from
    bool include_existing = 4;             // Include existing entries or only new ones
    string target_user = 5;               // Target user for routing (like other services)
}

message LogStreamResponse {
    string raw_line = 1;                  // Raw log line (unparsed)
    int64 line_number = 2;                // Line number in the file
    string stream_id = 3;                 // Unique stream identifier
    int64 file_position = 4;              // Position in file (for debugging)
    string detected_timestamp = 5;        // Simple timestamp detection (optional)
}

message StopStreamRequest {
    string stream_id = 1;
}

message StopStreamResponse {
    bool success = 1;
    string message = 2;
}